cg/meta/archive/ddn_dataflow.py

```diff
@@ -186,34 +186,39 @@ def auth_header(self) -> Dict[str, str]:
             self._refresh_auth_token()
         return {"Authorization": f"Bearer {self.auth_token}"}
 
-    def archive_folders(self, sources_and_destinations: Dict[Path, Path]) -> bool:
-        """Archives all folders provided, to their corresponding destination, as given by sources and destination parameter."""
-        transfer_data: List[TransferData] = [
-            TransferData(source=source.as_posix(), destination=destination.as_posix())
-            for source, destination in sources_and_destinations.items()
-        ]
-        transfer_request: TransferPayload = TransferPayload(files_to_transfer=transfer_data)
+    def archive_folders(self, sources_and_destinations: List[TransferData]) -> int:
+        """Archives all folders provided, to their corresponding destination,
+        as given by sources and destination parameter."""
+        transfer_request: TransferPayload = TransferPayload(
+            files_to_transfer=sources_and_destinations
+        )
         transfer_request.trim_paths(attribute_to_trim=SOURCE_ATTRIBUTE)
         transfer_request.add_repositories(
             source_prefix=self.local_storage, destination_prefix=self.archive_repository
         )
-        return transfer_request.post_request(
+        response: Response = transfer_request.post_request(
             headers=dict(self.headers, **self.auth_header),
             url=urljoin(base=self.url, url=DataflowEndpoints.ARCHIVE_FILES),
         )
+        if response.ok:
+            return response.json()["job_id"]
+        else:
+            raise IOError(response.text)
```

Comment:
There is a `response.raise_for_status()` which raises a `HTTPError` exception if the response has a non successful response code. What do you think?

```python
        response: Response = transfer_request.post_request(
            headers=dict(self.headers, **self.auth_header),
            url=urljoin(base=self.url, url=DataflowEndpoints.ARCHIVE_FILES),
        )
        response.raise_for_status()
        return response.json()["job_id"]
```

Reply:
Good idea! Implemented